cli/lsp/language_server.rs

```diff
@@ -427,21 +429,29 @@ impl Inner {
           import_map_str
         ))
       }?;
-      let import_map_path = import_map_url.to_file_path().map_err(|_| {
-        anyhow!("Cannot convert \"{}\" into a file path.", import_map_url)
-      })?;
-      info!(
-        "  Resolved import map: \"{}\"",
-        import_map_path.to_string_lossy()
-      );
-      let import_map_json =
+
+      let import_map_json = if import_map_url.scheme() == "data" {
+        get_source_from_data_url(&import_map_url)?.0
+      } else {
+        let import_map_path = import_map_url.to_file_path().map_err(|_| {
+          anyhow!("Cannot convert \"{}\" into a file path.", import_map_url)
+        })?;
+        info!(
+          "  Resolved import map: \"{}\"",
+          import_map_path.to_string_lossy()
+        );
         fs::read_to_string(import_map_path).await.map_err(|err| {
           anyhow!(
             "Failed to load the import map at: {}. [{}]",
             import_map_url,
             err
           )
-        })?;
+        })?
+      };
+      // TODO(@satyarohith): if import_map_url is a data url and import_map_json contains relative
+      // paths, we will get an Invalid address error in the logs because of the base_url not
+      // being a valid path. We should somehow support relative paths in import
+      // maps that are data urls.
```

Comment:
I would hope we addressed this on the command line @bartlomieju already when it was added to the CLI. Why don't we just use the cwd?

Reply:
I never thought of using data URLs for import maps. Unless there's something in the spec preventing usage of data URLs (I believe there's nothing about it) I'm fine with using CWD as a base url.

Reply:
Strangely, it appears we currently support them by accident, when we allowed remote import maps. 🤷

Reply:
I don't think that is what the spec suggests. Looking at https://wicg.github.io/import-maps/#sort-and-normalize-a-specifier-map it should either be the value of `--location`, or the `data:` url itself. If you try to resolve relative to the `data:` url this will fail, but the spec handles this gracefully (an error should be logged and the mapping should be set to "null").

Reply:
Then currently, we have a problem, as we accept data URLs for `--import-map` though we don't test for it.  Of course there is no cwd in a browser and we don't use `--location` for file based imports maps.

This PR just needs to deal with adding support for `data:` URLs in line with the CLI. @satyarohith can you take out this comment here, ensure the behaviour is aligned to another CLI and open an issue about data URLs for import maps having and undefined and untested behaviour all together?

Reply:
So should be prohibit usage of `data:` URLs for imports maps completely?

Reply:
No, I think @lucacasonato is saying that we shouldn't support relative specifiers in data URLs... we should log an error if we encounter one... and we should have tests for them.

Reply:
I'm a bit worried about the logging thing - some time ago we went away from logging warnings from import maps. Originally I implemented it strictly the the reference implementation including all the warning, but we went away from that. So the question is do we bring back other warnings/logs or just do the data URL one?

Reply:
What do we do now? Hard error?

Reply:
> What do we do now? Hard error?

No, the warnings are not displayed at all.

Reply:
>  we went away from logging warnings from import maps

Sorry for being ignorant; what was the reason for removing the logging?

Reply:
Satya, to be honest my memory doesn't serve me so well, but it rings a bell that we didn't want that output interleaved in tests when testing against WPT suite.

Reply:
> What do we do now? Hard error?

Yes, we do a hard error now.
```
➜ deno run --import-map="data:application/json;utf8,{\"imports\": { \"b\": \"./b.ts\" }}" mod.ts
Import map diagnostics:
  - Invalid address "./b.ts" for the specifier key "b".
error: Blocked by null entry for ""b""
```